### ISSUE TITLE
proposal tracking & viewing system

### ISSUE THREAD

**Issue by tilnow** [vgururao/yakcollective#68]
@vgururao 
I kinda like the website integration at proposal stage itself. If the process could create a draft project page that can change state from stub -- > proposed --> [greenlit | not greenlit] --> [in process | tabled ] -- > [staged] --> [published], with appropriate polling options at each stage if needed... that would be sweet

**Comment by tilnow** [comment]
i build some of the pieces, i think.
at https://inspiring-villani-8898ca.netlify.app/proposals/ i have a page that at build time pulls all the issues from  a github repository and shows them on a page. they could be organized by label, if we wish.
in github i built an automatic workflow (action) that (periodically) monitors reactions on issues and if they meet certain criteria, like number of +1 vs number of -1 adds a label. each label would correspond to a state of the proposal.
it is also possible to rebuild the page when a new label is added. this can also be done in javascript after build time.

**Comment by necopinus** [comment]
I'm a little reluctant to re-implement a view into the GitHub tracking system. Why not just use the issue tracker here?

If we want to pull data from Roam, why not have the approval process live there?

For example, I think that, if we use Roam for data, we will need to use namespaces to segregate "website" from "non-website" content. We could track the greenlight process by moving pages between different namespaces.

I think this is something worth talking through a bit "in person".

**Comment by tilnow** [comment]
my goal is not to duplicate github tracking, but rather to be able to view it on the web, which si freindlier interface than github or roam.
the question (as also noted by jenna dixon) is that we need to decide what is private and what not. viewing via website allows us to provide privacy via a members area or just by the link being private. same with viewing via knack or discord (though i do not think discord is good for viewing proposals)

**Comment by necopinus** [comment]
The website currently does not provide a private "members area", and I'm highly reluctant to create one. Layering on any kind of authentication is going to be easy to screw up, introduce _a lot_ of additional maintenance overhead, and reinvents the wheel.

Currently this repo and Roam are both public, and are both commonly used tools. I'm open to other tools, but I haven't yet seen a case where it makes sense for us to build our own.

**Comment by tilnow** [comment]
jenna says we can provide some of this functionality (without such coding) using knack.

**Comment by djinna** [comment]
hi @necopinus, welcome back, hope the conf was good. I hadn't given Knack a lot of thought when we spoke. Now see it as a quick way to stand up a safe member profile without hand coding even if we want to end up building our own down the road. Just offering, no push here. Do think we need to give folks granular control over privacy settings for any personal information we collect; eg knack would let us do public/yak only/admins only levels or similar. Connect also to Yak Map. Typing this on my phone sorry in advance for typos etc

**Comment by necopinus** [comment]
I'm pretty persuaded at this point that using a separate system for member data is the right way to go. I've created a separate issue, #71, to track that.

Once Gatsby is implemented (#12), we'll be pretty easily able to integrate any data source we can find a package for in NPM (though let's not go too crazy). I'm only going to support building out something new for this sort of proposal tracking system if we can't do it within the context of Roam + GitHub issues + Google Docs + whatever the member database is + ...

Leaving this open for now, but I think this is ticket is more about establishing a good _documented_ workflow, and _not_ about writing a new tool.

**Comment by tilnow** [comment]
so: lets build a prototype for member data on knack (@djinna - pls note)
regarding proposals, i am not sure i understand. we can:
1. maybe do via knack (@djinna - can we put together a prototype? including voting and data of project) - but i would still want to show in website
2. do via roam and discord (but that is an issue of tracking and usability - see the complaint of amazing stories)
3. do via github with only _display_ on the website (with problem you need a github login to vote, etc.)
4. some other tool - pls suggest

which do you think is best?

**Comment by necopinus** [comment]
Probably depends on #83 and/or #71.
